src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1207 @@
+
+        # do a depth-first search
+        while stack:
+            i = stack.pop()
+            for k in range(n):
+                j = i.mutate(k, inplace=False)
+                # need good convenient method of representing seeds on graph
+                covers.append((frozenset(i.cluster()), frozenset(j.cluster())))
+                if j not in known_seeds:
+                    known_seeds += [j]
+                    stack.append(j)
+        G = Graph(covers)
+        G.relabel()
+        return G
+
+    def cluster(self):
+        r"""
+        Return the cluster variables of ``self``.
+
+        EXAMPLES:
+
+        Get the cluster variables after performing a mutation::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1})
+            sage: S.mutate(0)
+            sage: S.cluster()
+            ((x2 + 1)/x1, x2)
+        """
+        return tuple(self._cluster_vars)
+
+    def exchange_polys(self):
+        r"""
+        Return the exchange polynomials of ``self``.
+
+        EXAMPLES:
+
+        Get the exchange polynomials after performing a mutation::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 3 + 4*x2, x2: 5 + 6*x1})
+            sage: S.mutate(0)
+            sage: S.exchange_polys()
+            [4*x2 + 3, 5*x1 + 18]
+        """
+
+        return list(self._exchange_polys)
+
+    def laurent_polys(self):
+        r"""
+        Return the exchange Laurent polynomials of ``self``.
+
+        EXAMPLES:
+
+        This seed has non-trivial exchange Laurent polynomials::
+
+            sage: var('x1, x2, x3')
+            (x1, x2, x3)
+            sage: S = LPASeed({x1: 1 + x2 + x3, x2: 1 + x3, x3: 1 + x2})
+            sage: S.laurent_polys()
+            [(x2 + x3 + 1)/(x2*x3), x3 + 1, x2 + 1]
+        """
+
+        return list(self._laurent_polys)
+
+    def rank(self):
+        r"""
+        Return the rank of ``self``. This is the number of
+        cluster variables in ``self``.
+
+        EXAMPLES:
+
+        The rank is the number of cluster variables in the seed::
+
+            sage: var('x1,x2,x3,x4')
+            (x1, x2, x3, x4)
+            sage: LPASeed({x1: 1 + x2, x2: 1 + x1, x3: 1 + x4, x4: 1 + x1}).rank()
+            4
+        """
+
+        return self._rank
+
+    def randomly_mutate(self, depth, inplace=True):
+        r"""
+        Randomly mutates this seed at ``depth`` indices. Useful for working out
+        if a seed produces a finite type LP algebra.
+
+        INPUT:
+
+        - ``depth`` -- integer, the number of random mutations to perform.
+
+        EXAMPLES:
+
+        We suspect this seed is infinite type, so we perform some random
+        mutations::
+
+            sage: var('x1,x2,x3')
+            (x1, x2, x3)
+            sage: S = LPASeed({x1: 1 + x2*x3, x2: 1 + x1, x3: 1 + x2})
+            sage: S.randomly_mutate(7)
+            sage: print(S)  # random
```

Comment:
This is not a good test. You should check something to make sure that it worked correctly, say length of `S.mutation_sequence()`.